### PR TITLE
feat: adds alert task message and common consumer and producer for it

### PR DESCRIPTION
**Pull request by kotharironak**
## Description
As part of addressing ticket - hypertrace/hypertrace#251, this PR adds,
- alert task avdl message
- Kafka based consumer and producer for sending messages to Kafka queue

### Testing
Was done as part of this PR - https://github.com/hypertrace/hypertrace-alert-engine/pull/8

Note: Will generalize the queue interface as follow-up PR. This one will unblock the first iteration for the producer and evaluator side of the work.

### CODE REVIEW

File: metric-anomaly-task-manager/src/main/resources/configs/common/application.conf [hunk 1]
```diff
@@ -5,3 +5,4 @@ main.class = org.hypertrace.alert.engine.metric.anomaly.task.manager.MetricAnoma
 
 logger.names = ["file"]
 logger.file.dir = "/var/logs/metric-anomaly-rule-manager"
+
```
Comment (kotharironak) [inline metric-anomaly-task-manager/src/main/resources/configs/common/application.conf:8]:
```suggestion
```

**Comment by kotharironak** [comment]
@JBAhire When I locally run with -x avroCompatibilityCheck opiton, I am able to skip this task. Is the current action doesn't support this?

**Comment by kotharironak** [comment]
Sample configuration will look something as
```
queue.config {
  type = "kafka"
  kafka = {
    topic = "alert-tasks"
    bootstrap.servers = "localhost:9093"
    producer = {

    }
    consumer = {

    }
  }
}
```

File: metric-anomaly-data-model/src/main/java/org/hypertrace/alert/engine/metric/anomaly/datamodel/queue/KafkaAlertTaskConsumer.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  public KafkaAlertTaskConsumer(Config kafkaQueueConfig) {
+    this.kafkaConfigReader = new KafkaConfigReader(kafkaQueueConfig);
+    Properties props = new Properties();
+    props.putAll(kafkaConfigReader.getConsumerConfig(createBaseProperties()));
+    consumer = new KafkaConsumer<String, ByteBuffer>(props);
+    consumer.subscribe(Collections.singletonList(kafkaConfigReader.getTopicName()));
+  }
+
+  public AlertTask dequeue() throws IOException {
```
Comment (rish691) [inline metric-anomaly-data-model/src/main/java/org/hypertrace/alert/engine/metric/anomaly/datamodel/queue/KafkaAlertTaskConsumer.java:36]: we could have removed this method as well and renamed the class to KafkaConsumer

Comment (kotharironak) [reply metric-anomaly-data-model/src/main/java/org/hypertrace/alert/engine/metric/anomaly/datamodel/queue/KafkaAlertTaskConsumer.java:36]: Well, this is not a general consumer, for now, it is producing alert task and consuming it. We will convert to queue interface as follow.

**Review by rish691 — APPROVED** [review]

**Pull request merged**